src/hii_road.py

```diff
@@ -1,358 +1,172 @@
+        self.road_direct = None
+
+    def kernel(self):
+        return ee.Kernel.euclidean(radius=self.KERNEL_DISTANCE, units="meters")
+
+    def osm_groads_combined_influence(self):
+        osm_band_names = self.osm.bandNames()
+        road_band_names = list(self.weights["osm"].keys())
+        road_bands = osm_band_names.filter(ee.Filter.inList("item", road_band_names))
+        osm_roads = self.osm.select(road_bands)
+        weights_image = ee.Dictionary(self.weights["osm"]).toImage().select(road_bands)
+
+        # creates 1km wide images with direct influence values
+        osm_direct = (
+            osm_roads.distance(kernel=self.kernel(), skipMasked=False)
+            .lte(self.DIRECT_INFLUENCE_RADIUS)
+            .selfMask()
+            .multiply(weights_image)
+        )
+        groads_direct = (
+            self.groads.distance(kernel=self.kernel(), skipMasked=False)
+            .lte(self.DIRECT_INFLUENCE_RADIUS)
+            .selfMask()
+            .multiply(self.weights["groad"])
+        )
+
+        # masks groads that overlap (< 500m) from an OSM road
+        non_osm_binary = osm_direct.reduce(ee.Reducer.anyNonZero()).unmask(0).eq(0)
+        groads_fill = groads_direct.updateMask(non_osm_binary)
+
+        self.road_direct = (
+            osm_direct.addBands(groads_fill)
+            .reduce(ee.Reducer.sum())
+            .multiply(2)
+            .unmask(0)
+            .rename("road_direct_influence")
+        )
+
+    def groads_influence(self):
+        self.road_direct = (
+            (
+                self.groads.distance(kernel=self.kernel(), skipMasked=False)
+                .lte(self.DIRECT_INFLUENCE_RADIUS)
+                .selfMask()
+                .multiply(self.weights["groad"])
+            )
+            .multiply(2)
+            .unmask(0)
+            .rename("road_direct_influence")
         )
 
-        highway_turning_circle, highway_turning_circle_date = self.get_most_recent_image(
-            ee.ImageCollection(self.inputs["highway_turning_circle"]["ee_path"])
+    def calc(self):
+        if self.osm:
+            self.osm_groads_combined_influence()
+        else:
+            self.groads_influence()
+        road_binary = self.road_direct.gt(0).unmask(0)
+        road_indirect = (
+            road_binary.eq(0)
+            .cumulativeCost(road_binary, self.INDIRECT_INFLUENCE_RADIUS)
+            .multiply(self.DECAY_CONSTANT)
+            .exp()
+            .multiply(self.INDIRECT_INFLUENCE)
+            .reproject(crs=self.crs, scale=self.scale)
+            .unmask(0)
+            .rename("roads_indirect_influence")
         )
-
-        highway_unclassified, highway_unclassified_date = self.get_most_recent_image(
-            ee.ImageCollection(self.inputs["highway_unclassified"]["ee_path"])
+        # TODO: determine if drivers are to be normalized or not.
+        # TODO: values beyond indirect range (15km) - do we set to 0 or mask?
```

Comment:
Good question -- my take is, they should be 0 because those cells have 0 influence contributing to HF coming from roads (not that they're nonexistent). But cells outside the water mask should be masked.

Reply:
water is masked. Values beyond 15km are 0 where they fall on land.